### PR TITLE
Jitpack support

### PR DESCRIPTION
**Pull request by almozavr**
- Added android-maven-plugin to support jitpack.io
- .gitignore modified to be more ide-independant

### CODE REVIEW

**Comment by roman-mazur** [comment]
What is the motivation? Any problems with maven?

**Comment by almozavr** [comment]
There are two main points:
1. Snapshot-independance - in case there are some badly-needed commits
exist, but not yet released;
2. Easier fork-based releases (no need to hustle with own maven, etc)

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,9 @@ local.properties
 # Gradle
 build
 .gradle
+
+# Android Studio
+.idea/*
```
Comment (roman-mazur) [inline .gitignore]: We tend to omit IDE-related stuff in project `.gitignore`. Number of IDEs or editors people can use to edit code is not limited. Hence, theoretically we cannot cover all the cases here. `.gitignore` rules in this files are related to the build system used in the project. And IDE-related stuff is recommended to be kept in [global git ignore configuration](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) on your machine to fit your preferences and avoid copying it in all the projects.

**Comment by roman-mazur** [comment]
We didn't configure automatic snapshot publication after extracting this project to a separate repo. I will do it today. SNAPSHOTs will be published on every push to master (pull request merge).
Jitpack relies on creating GitHub releases, which is not something we are ready to commit to maintain :)
We support `CHANGELOG` file and create tags on every release. But we do not create release pages on github.
Hence, this will be closed after snapshots publication is configured.

**Comment by almozavr** [comment]
> Jitpack relies on creating GitHub releases
The magick (for forks and non-patient contributors) is `Jitpack` works with commit-sha as well, so is a major plus to have. Also, it has no impact on code, compiling or other. Am I miss something?

**Pull request closed without merging**